### PR TITLE
Fix partial test failure handling

### LINKED ISSUE #14: Test execution does not terminate on partial test fail
When test execution failed in the middle of the test, test-runner does not terminate test execution.

Test-runner should terminate test on partial test failure.
And, reset WebDriver session to continue new test-suite execution.

WebDriver session reset should occur in these timing.
- At the beginning of tests execution with persistSession disabled
- At the beginning of test-suite execution after a test-command failure

### PR DESCRIPTION
**Pull request by yaamai**
Fix #14

### CODE REVIEW

File: side_runner_py/main.py [hunk 1]
```diff
@@ -66,46 +79,108 @@ def _call_hook_script(pattern):
...
+        logger.debug('Leave tests {}'.format(test_id))
+    except Exception:
+        traceback_msg = traceback.format_exc()
+        logger.warning(traceback_msg)
+
+        # close driver if exception or test-failure occur in tests session
+        _close_driver_or_skip(driver)
+
+    # close driver if test_suite require session close
+    if not test_suite.get('persistSessiona', False):
```
Comment (rsp9u) [inline side_runner_py/main.py]: typo

File: tests/unit/test_main.py [hunk 2]
```diff
@@ -24,21 +24,92 @@ def test_main_with_glob_no_match(mocker):
...
+    mocker.patch('side_runner_py.main.with_retry')
+    mocker.patch('side_runner_py.main.get_screenshot')
+    mocker.patch('side_runner_py.main.execute_test_command').return_value = {'is_failed': False}
+    driver_close = mocker.patch('side_runner_py.main._close_driver_or_skip')
+
+    # parepare mock test file
+    sidefile_a = tmp_path / "a.json"
+    orig_test_project_a = {
+        'id': 'foobar_a',
+        'suites': [{'name': 'foobar_a', 'id': 'foobar_a', 'tests': ['foobar_a'], 'persistSessiona': True}],
```
Comment (rsp9u) [inline tests/unit/test_main.py]: typo (Sessiona)

File: tests/unit/test_main.py [hunk 3]
```diff
@@ -24,21 +24,92 @@ def test_main_with_glob_no_match(mocker):
...
+    mocker.patch('side_runner_py.main.with_retry')
+    mocker.patch('side_runner_py.main.get_screenshot')
+    mocker.patch('side_runner_py.main.execute_test_command').return_value = {'is_failed': False}
+    driver_close = mocker.patch('side_runner_py.main._close_driver_or_skip')
+
+    # parepare mock test file
+    sidefile_a = tmp_path / "a.json"
+    orig_test_project_a = {
+        'id': 'foobar_a',
+        'suites': [{'name': 'foobar_a', 'id': 'foobar_a', 'tests': ['foobar_a'], 'persistSessiona': False}],
```
Comment (rsp9u) [inline tests/unit/test_main.py]: typo (Sessiona)

File: tests/unit/test_main.py (hunk 2 above)
Comment (yaamai) [reply tests/unit/test_main.py]: fixed

File: side_runner_py/main.py (hunk 1 above)
Comment (yaamai) [reply side_runner_py/main.py]: fixed

File: tests/unit/test_main.py (hunk 3 above)
Comment (yaamai) [reply tests/unit/test_main.py]: fixed

**Review by rsp9u — APPROVED** [review]
LGTM

**Pull request merged**